### ISSUE TITLE
Server freezes when clipboard contains an image

### ISSUE THREAD

**Issue by nneul** [symless/synergy-core#6409]
### Operating Systems ###

Ubunty 16 client and Ubuntu 18 server, both x86_64

### Synergy Version ###

synergys 1.10.0, protocol version 1.6

### Steps to Reproduce ###

Screenshot or similar to copy an image to the X clipboard on the server. It appears to be size based, smaller copies do not appear to trigger the same lockup.

### Log URLs ###

No indications of failure on the client side, only on server. The last messages in the log are it attempting to retrieve text/html and then image/bmp. 

[2018-09-04T14:59:18] DEBUG1: request succeeded after 0.000001s
[2018-09-04T14:59:18] DEBUG:   available targets: TARGETS (450), MULTIPLE (451), image/tiff (857), image/jpeg (660), image/x-MS-bmp (848)
[2018-09-04T14:59:18] DEBUG1: request selection=CLIPBOARD (448), target=text/html (574), window=5000004
[2018-09-04T14:59:18] DEBUG2: can't read property 454 on window 0x05000004
[2018-09-04T14:59:18] DEBUG1: request failed after 0.000003s
[2018-09-04T14:59:18] DEBUG1: can't get data for selection target text/html (574)
[2018-09-04T14:59:18] DEBUG1:   no data for target text/html (574)
[2018-09-04T14:59:18] DEBUG1: request selection=CLIPBOARD (448), target=image/bmp (575), window=5000004
[2018-09-04T14:59:18] DEBUG2: read property 454 on window 0x05000004: bytes=4
[2018-09-04T14:59:25] DEBUG2: reading secure socket

Now that I know how to prevent it, it's easier to deal with.

**Comment by nneul** [comment]
Does not appear to change behavior if I change the size limit on clipboard sharing.

**Comment by nneul** [comment]
And no change with latest 1.10.1 build.

**Comment by nneul** [comment]
Little bit of brute force debug - it's getting stuck oscillating between 1343 and 1348 debug lines, but only when I move the mouse - it's not a busy spin:

src/lib/platform/XWindowsClipboard.cpp:

```1341         if (noWait || XPending(display) > 0) {
1342             while (!m_done && !m_failed && (noWait || XPending(display) > 0)) {
1343 LOG((CLOG_DEBUG1 "dbg %s:%d", __FILE__, __LINE__));
1344                 XNextEvent(display, &xevent);
1345                 if (!processEvent(display, &xevent)) {
1346                     // not processed so save it
1347                     events.push_back(xevent);
1348 LOG((CLOG_DEBUG1 "dbg %s:%d", __FILE__, __LINE__));
1349                 }
1350                 else {
1351                     // reset timer since we've made some progress
1352                     timeout.reset();
```

It's not making it out of the while loop on 1342. It looks like it's almost got stuck with reprocessing the same event every time and sticking it back on the queue.

**Comment by nneul** [comment]
A trivial fix for this was provided, when can we expect it to get merged or responded to?

**Comment by Jnewbon** [comment]
Fixed by PR #6410
